### ISSUE TITLE
Failed to build curl on CentOS 7

### ISSUE THREAD

**Issue by phxyz12** [spack/spack#7943]
Summary

### Expected result:

`$ spack install curl`

installs a working installation of curl.

### What happened instead:

```$ spack install curl leads to a very early error:

==> Executing phase: 'configure'
==> Error: AttributeError: 'Curl' object has no attribute 'satisfies'
AttributeError: AttributeError: 'Curl' object has no attribute 'satisfies'
/cfs/home/swadm/spack/var/spack/repos/builtin/packages/curl/package.py:78, in configure_args:
1             spec = self.spec
2     
3             args = ['--with-zlib={0}'.format(spec['zlib'].prefix)]
>>> 4             if self.satisfies('+darwinssl'):
5                 args.append('--with-darwinssl')
6             else:
7                 args.append('--with-ssl={0}'.format(spec['openssl'].prefix))

See build log for details:
  /cfs/home/swadm/spack/var/spack/stage/curl-7.59.0-uoqifrmsrxjm6s5ur2zlvsua3rcmlxal/curl-7.59.0/spack-build.out
```
The file spack-build.out only contains the lines:
```
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
```

### Steps to reproduce the issue

1. Clone current spack repo
2. Try a spack install curl

### How to make it work again (just a workaround)

When I use the little older package.py file which ends with curl version 7.56.0, the installation works. The package.py file changed recently and now it seems to be broken in my system.

### Information on your system

CentOS 7, gcc 7.3.0

**Comment by healther** [comment]
This was introduced by me in 6d1097f and fixed in c8e3c94, either cherry-pick the latter or upgrade to the current spack HEAD. Sorry for the inconvenience

**Comment by phxyz12** [comment]
Hi
thanks for the quick reply. I wasn't aware that it's already fixed. I close the issue.
Best, Pascal
